attachment_delete_restrict/models/ir_attachment.py

```diff
@@ -64,15 +64,16 @@ def _check_custom_delete_attachment(self, model=None, allow_owner_and_admin=Fals
             return self._raise_delete_attachment_error(allowed_users)
 
     def unlink(self):
-        res_models = list(set(self.filtered("res_model").mapped("res_model")))
-        if res_models:
-            models = self.env["ir.model"].search([("model", "in", res_models)])
-            name2models = {m.model: m for m in models}
-            for rec in self:
-                if rec.res_model:
-                    model = name2models[rec.res_model]
-                    if model.restrict_delete_attachment == "default":
-                        rec.sudo()._check_delete_attachment()
-                    else:
-                        rec.sudo()._check_delete_attachment(model)
+        if not self.env.su:
+            res_models = list(set(self.filtered("res_model").mapped("res_model")))
+            if res_models:
+                models = self.env["ir.model"].search([("model", "in", res_models)])
+                name2models = {m.model: m for m in models}
+                for rec in self:
+                    if rec.res_model:
+                        model = name2models[rec.res_model]
+                        if model.restrict_delete_attachment == "default":
+                            rec.sudo()._check_delete_attachment()
+                        else:
+                            rec.sudo()._check_delete_attachment(model)
```

Comment:
Could we please reduce the diff?
```suggestion
        if self.env.su:
            return super().unlink()
        res_models = list(set(self.filtered("res_model").mapped("res_model")))
        if res_models:
            models = self.env["ir.model"].search([("model", "in", res_models)])
            name2models = {m.model: m for m in models}
            for rec in self:
                if rec.res_model:
                    model = name2models[rec.res_model]
                    if model.restrict_delete_attachment == "default":
                        rec.sudo()._check_delete_attachment()
                    else:
                        rec.sudo()._check_delete_attachment(model)
```